### DISCUSSION TITLE
[General] <Title>How to implement (EIP-1559) transactions on the zkaync era chain using web3.py

### DISCUSSION THREAD

**Discussion by baalisgood** [zkSync-Community-Hub/zksync-developers#275] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

What happened? When I interact with a dapp front end, my transactions look like this:
Txn Type: 2 (EIP-1559), when I interact with web3.py, my transaction looks like this:
Txn Type: 0 (Legacy)

### What is the specific issue or error you're encountering?

web3.py transaction issues

### Can you share the error messages or logs you're receiving, if any?

(https://era.zksync.network/tx/0xbe6a547ad6652708097df1929f98de224ecf4e2ef9ec3dd58bb5d5c3d5290c6d)This is my transaction using web3.py，(https://era.zksync.network/tx/0xc3f721134bb7d826913161df9490ccd48b2e3d5d17b3c59140e38a0f14a93a72)This is my hash using front-end transactions

### Have you made any recent changes to the contract before encountering this issue?

There have been no recent changes to the code, but I have seen some updates, although I am not familiar with them

### Are there any external libraries or contracts that your contract interacts with?

no

### Can you provide the relevant portions of your contract code where the issue is occurring?

no

### Have you tried to isolate the problem, and if so, what were the results?

no

### What steps have you already taken to try to resolve the issue?

dict_transaction

**Comment by dutterbutter** [comment, marked as answer]
@wt195799611 we have our own dedicated python sdk: https://github.com/zksync-sdk/zksync2-python can you please try using it as its derived from web3.py for zkSync.

**Comment by candacekrank** [comment]
It seems like you're encountering discrepancies in transaction types between interacting with a DApp frontend and using web3.py. This could be due to differences in how transactions are constructed or signed between the two methods. To resolve this issue, you may need to investigate and ensure consistency in how transactions are handled across both methods. Review the documentation and code for both the DApp frontend and web3.py to identify any differences in transaction construction or signing processes. Additionally, consider reaching out to the developers or community forums for assistance in troubleshooting and resolving this discrepancy.

**Comment by sarportzapbisusan** [comment]
It seems there is a discrepancy in the transaction type (EIP-1559 vs. Legacy) when interacting with the dapp frontend and web3.py. Further investigation is needed to understand and resolve this issue.

**Comment by lykinsjennifer** [comment]
It seems like there may be a discrepancy in transaction types between interacting with the dapp front end and using web3.py. Ensure that both methods are using the same transaction type (EIP-1559 or Legacy) to avoid inconsistencies.

**Comment by colleennealv** [comment]
It seems that you are encountering a difference in transaction types when interacting with a dapp front end versus using web3.py. This could be due to different transaction signing methods or configurations between the two approaches. Please ensure consistency in transaction signing methods and configurations to avoid such discrepancies.

**Comment by thakurmatt** [comment]
The user is encountering an issue where transactions made through a dapp front end show up as EIP-1559 type, while transactions made through web3.py show up as Legacy type. The user has not made any recent changes to the contract, and there are no external libraries or contracts involved. No relevant portions of contract code are provided, and the user has not attempted to isolate or resolve the issue yet. No repository link or additional details are provided.
